llvm/lib/IR/Instructions.cpp

```diff
@@ -60,22 +61,33 @@ static cl::opt<bool> DisableI2pP2iOpt(
 std::optional<TypeSize>
 AllocaInst::getAllocationSize(const DataLayout &DL) const {
   TypeSize Size = DL.getTypeAllocSize(getAllocatedType());
-  if (isArrayAllocation()) {
-    auto *C = dyn_cast<ConstantInt>(getArraySize());
-    if (!C)
-      return std::nullopt;
-    assert(!Size.isScalable() && "Array elements cannot have a scalable size");
-    Size *= C->getZExtValue();
+  if (!isArrayAllocation()) {
```

Comment:
Why do you need to invert the condition?

Reply:
early exit for better readability

Reply:
I would prefer to keep the existing code style and only fix the overflow issue.

Reply:
Sure, it's my first time contributing to llvm, so I was just trying to conform to the coding standard which says to use early exit to minimize nested if's.

Reply:
The diff becomes smaller and it is easier to review when you focus on the overflow check.